lib/ansible/modules/storage/glusterfs/gluster_peer.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2015 Nandaja Varma <nvarma@redhat.com>
+# Copyright 2018 Red Hat, Inc.
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: gluster_peer
+short_description: Attach/Detach peers to/from the cluster
+description:
+  - Create or diminish a GlusterFS trusted storage pool. A set of nodes can be
+    added into an existing trusted storage pool or a new storage pool can be
+    formed. Or, nodes can be removed from an existing trusted storage pool.
+version_added: "2.6"
+author: Sachidananda Urs (@sac)
+options:
+    state:
+       choices: ["present", "absent"]
+       description:
+          - Determines whether the nodes should be attached to the pool or
+            removed from the pool. If the state is present, nodes will be
+            attached to the pool. If state is absent, nodes will be detached
+            from the pool.
+       required: true
+    nodes:
+       description:
+          - List of nodes that have to be probed into the pool.
+       required: true
+    force:
+       type: bool
+       default: "false"
+       description:
+          - Applicable only while removing the nodes from the pool. gluster
+            will refuse to detach a node from the pool if any one of the node
+            is down, in such cases force can be used.
+requirements:
+  - GlusterFS > 3.2
+notes:
+  - This module does not support check mode.
+'''
+
+EXAMPLES = '''
+- name: Create a trusted storage pool
+  gluster_pool:
+        state: present
+        nodes:
+             - 10.0.1.5
+             - 10.0.1.10
+
+- name: Delete a node from the trusted storage pool
+  gluster_pool:
+         state: absent
+         nodes:
+              - 10.0.1.10
+
+- name: Delete a node from the trusted storage pool by force
+  gluster_pool:
+         state: absent
+         nodes:
+              - 10.0.0.1
+         force: true
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ast import literal_eval
+
+
+class Peer(object):
+    def __init__(self, module):
+        self.module = module
+        self.state = self._validated_params('state')
```

Comment:
`state` parameter is already flagged as `required`, it can not be `None` (only `present` or `absent`). Then call to `self._validated_params` must be removed and `self.state = self.module.params['state']` should be used.

Reply:
Ack! Done.